packages/launchpad-auth/index.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+/**
+ * Utility functions for generating and validating Launchpad tokens
+ *
+ * @module launchpad-auth
+ */
+
+const path = require('path');
+const pick = require('lodash.pick');
+const {
+  getS3Object,
+  s3ObjectExists,
+  s3PutObject
+} = require('@cumulus/aws-client/S3');
+const Logger = require('@cumulus/logger');
+
+const LaunchpadToken = require('./LaunchpadToken');
+
+const log = new Logger({ sender: '@cumulus/launchpad-auth' });
+
+/**
+ * Get S3 location of the Launchpad token
+ *
+ * @returns {Promise.<Object.<string, string>>} - S3 Bucket and Key where Launchpad token is stored
+ *
+ * @private
+ */
+function launchpadTokenBucketKey() {
+  if (!(process.env.stackName && process.env.system_bucket)) {
+    throw new Error('must set environment variables process.env.stackName and process.env.system_bucket');
+  }
+
+  return {
+    Bucket: process.env.system_bucket,
+    Key: path.join(process.env.stackName, 'launchpad/token.json')
+  };
+}
+
+/**
+ * Retrieve Launchpad token from S3
+ *
+ * @returns {Promise.<string>} - the Launchpad token, null if token doesn't exist or invalid
+ *
+ * @async
+ * @private
+ */
+async function getValidLaunchpadTokenFromS3() {
+  const s3location = launchpadTokenBucketKey();
+  const keyExists = await s3ObjectExists(s3location);
+
+  let token = null;
+  if (keyExists) {
+    const s3object = await getS3Object(s3location.Bucket, s3location.Key);
+    const launchpadToken = JSON.parse(s3object.Body.toString());
+
+    // check if token is still valid
+    if (Date.now() / 1000 < launchpadToken.session_maxtimeout + launchpadToken.session_starttime) {
+      token = launchpadToken.sm_token;
+    }
+  }
+
+  return token;
+}
+
+/**
+ * Get a Launchpad token
+ *
+ * @param {Object} params - the configuration parameters for creating LaunchpadToken object
+ * @param {string} params.api - the Launchpad token service api endpoint
+ * @param {string} params.passphrase - the passphrase of the Launchpad PKI certificate
+ * @param {string} params.certificate - the name of the Launchpad PKI pfx certificate
+ *
+ * @returns {Promise.<string>} - the Launchpad token
+ *
+ * @async
+ * @alias module:launchpad-auth
```

Comment:
Hmm, TIL.    Interesting.

Reply:
Yeah, this was necessary for the type of the functions in the generated API docs to be `exported function`, which is what they are